src/process/message_processor.go

```diff
@@ -1,48 +1,101 @@
 package process
 
 import (
+	"BGPAlert/analyze"
 	"BGPAlert/common"
-	"errors"
+	"BGPAlert/config"
 	"fmt"
+	"strconv"
 	"time"
 )
 
-// Constantly reads channel of messages and stores them in Window objects to send through windowChannel for analysis
-func ProcessBGPMessages(msgChannel chan common.BGPMessage, windowChannel chan common.Window) error {
-	var bucketMap = make(map[time.Time][]common.BGPMessage)
+// Constantly read messages from channel, build up windows with frequency maps, calling analysis on windows when full
+func ProcessBGPMessages(msgChannel chan common.BGPMessage, config *config.Configuration) error {
+	// Parse windowSize from config
+	var maximumBuckets int
+	// If windowSize can't be parsed to an int -> just default to 30
+	if windowSize, err := strconv.ParseInt(config.WindowSize, 10, 64); err != nil {
```

Comment:
Do we need to establish a upper and min window size bound? Is there a bucket size that is TOO big?

Reply:
I think it's fine to give the user the option to just chose whatever they want to test, but this could be brought up in the future if we have trouble storing too many buckets.